### DISCUSSION TITLE
Is it possible to just obtain the descriptors for debugging purposes?

### DISCUSSION THREAD

**Discussion by hyhuang00** [deepmodeling/deepmd-kit#2536] [Q&A]
If it's possible, what kind of function should I use to obtain the raw descriptors before any neural network layers?

**Comment by njzjz** [comment]
Unless you write your function like `deepmd.infer.DeepPot.eval_descriptor`. 

https://github.com/deepmodeling/deepmd-kit/blob/3ac8c4c77b61ee4c645cfb59eaefb104d73c0309/deepmd/infer/deep_pot.py#L577-L579
